app/controllers/comments_controller.rb

```diff
@@ -314,7 +314,16 @@ def permit_commenter
   def user_blocked?
     return false if current_user.blocked_by_count.zero?
 
-    UserBlock.blocking?(@comment.commentable.user_id, current_user.id) ||
-      UserBlock.blocking?(@comment.parent&.user_id, current_user.id)
+    blocked_by_commentable_author = UserBlock.blocking?(@comment.commentable.user_id, current_user.id)
+    blocked_by_comment_author = @comment.parent && UserBlock.blocking?(@comment.parent.user_id, current_user.id)
+
+    blocked_by_commentable_author || blocked_by_comment_author || blocker_upthread?(@comment.parent)
+  end
+
+  def blocker_upthread?(comment)
```

Comment:
Optional suggestion:
Adding a comment on this method.

```
# This method checks if any user who has commented above the given comment in
# the thread (i.e., any of the comment's ancestors) has blocked the current user.
```

Reply:
@dukegreene You can ignore this comment. I have suggested same comment before but you can directly make a commit of that, if you think that's correct else you can ignore this. Anyway its fine. Thanks.